### ISSUE TITLE
consumer.Http's WithHeaders truncates message body

### ISSUE THREAD

**Issue by mhils** [trivago/gollum#70]
Steps to reproduce the problem:

1. Use the following gollum config:
```
- "consumer.Http":
    Address: "localhost:9090"
    WithHeaders: true

- "producer.Console":
    Formatter: "format.Envelope"
    EnvelopePrefix: "[start]"
    EnvelopePostfix: "[end]"
```
2. Issue the following HTTP request:
```
wget https://uploads.maximilianhils.com/2017-03/repro
cat repro | nc localhost 9090
```

3. Observe that output correctly ends with "<<<":

```
$ gollum -c repro.conf
[start][snip http headers]>>>>>>>>>>>>[snip content]<<<<<<<<<<<[end]
```

4. Set `WithHeaders` to `false`, restart gollum
5. Issue the same HTTP request again. The message body is now truncated:
```
$ gollum -c repro.conf
[start]>>>>>>>>>>>>[snip content]NoZW1hIiB4bWxuczpjd21wPSJ1cm4[end]
```

**Comment by arnecls** [comment]
A few questions:
- Which gollum version did you use? 
- If 0.4.4, does it also happen with 0.4.5 (branch v0.4.5dev)?
- If you compiled it yourself, which go version did you use to compile it?

**Comment by mhils** [comment]
> Which gollum version did you use?

I initially used the binaries from https://github.com/trivago/gollum/releases/download/v0.4.4/gollum_win.zip when I reported this. I have just tested all combinations of {master, v0.4.5dev and v0.5.0dev} x {go 1.6 or go 1.8}, they all exhibit this issue for me.

The output is cut off after 863 bytes. HTTP headers are 161 bytes, so the TCP stream is cut off at exactly 1024 bytes. Looking at WireShark, I see that my netcat sends TCP packets with 1024 bytes each. I'm no expert in Go, but it seems like [`consumer/http.go#L171`](https://github.com/trivago/gollum/blob/20bbc74df112cfded2d102b339158f91b297c252/consumer/http.go#L171) probably returns before reading the buffer in full. I'll send a PR to fix that in a second. :)
